controllers/clusterlogging/clusterlogging_controller.go

```diff
@@ -118,23 +84,112 @@ func (r *ReconcileClusterLogging) Reconcile(ctx context.Context, request reconci
 		return result, err
 	}
 
-	return reconcileResult, err
+	return ctrl.Result{}, err
 }
 
-func (r *ReconcileClusterLogging) updateStatus(instance *loggingv1.ClusterLogging) (reconcile.Result, error) {
+func (r *ReconcileClusterLogging) updateStatus(instance *loggingv1.ClusterLogging) (ctrl.Result, error) {
 	if err := r.Client.Status().Update(context.TODO(), instance); err != nil {
 		telemetry.Data.CLInfo.Set("healthStatus", constants.UnHealthyStatus)
 		telemetry.UpdateCLMetricsNoErr()
 		log.Error(err, "clusterlogging-controller error updating status")
-		return reconcileResult, err
+		return ctrl.Result{}, err
+	}
+
+	return ctrl.Result{}, nil
+}
+
+func IsElasticsearchRelatedObj(obj client.Object) (bool, ctrl.Request) {
+	if obj.GetNamespace() == "" {
+		// ignore cluster scope objects
+		return false, ctrl.Request{}
+	}
+
+	if constants.OpenshiftNS != obj.GetNamespace() {
+		// ignore object in another namespace
+		return false, ctrl.Request{}
+	}
+
+	if value, exists := obj.GetLabels()["cluster-name"]; !exists || value != constants.ElasticsearchName {
+		return false, ctrl.Request{}
+	}
+
+	if value, exists := obj.GetLabels()["component"]; !exists || value != constants.ElasticsearchName {
+		return false, ctrl.Request{}
 	}
```

Comment:
IMHO this can be simplified by checking only `obj.Name != constants.ElasticsearchName`. Essentially you want the last `return` to return the custom `ctrl.Request{...}` object if the event object has the name `elasticsearch` in the `openshift-logging` namespace.